examples/twittermap/web/public/javascripts/map/heatmap.js

```diff
@@ -76,25 +77,47 @@ angular.module("cloudberry.map")
           field: ["id", "coordinate", "place.bounding_box", "create_at", "user.id"]
         },
         option: {
-          sliceMillis: cloudberryConfig.querySliceMills
+          sliceMillis: cloudberryConfig.querySliceMills,
+          returnDelta: true
         }
       };
 
-      var heatTimeJson = queryUtil.getTimeBarRequest(cloudberry.parameters, $scope.geoIdsNotInTimeSeriesCache);
-
-      cloudberryClient.send(heatJson, function(id, resultSet, resultTimeInterval){
-        if(angular.isArray(resultSet)) {
-          cloudberry.commonTweetResult = resultSet[0].slice(0, queryUtil.defaultSamplingSize - 1);
-          cloudberry.heatmapMapResult = resultSet[0];
-        }
-      }, "heatMapResult");
 
       // Complete time series cache hit case - exclude time series request
       if($scope.geoIdsNotInTimeSeriesCache.length === 0) {
         cloudberry.commonTimeSeriesResult = TimeSeriesCache.getTimeSeriesValues(cloudberry.parameters.geoIds, cloudberry.parameters.geoLevel, cloudberry.parameters.timeInterval);
+
+        // Partial heatmap cache hit hit case
+        if (!heatMapResultCache.cacheIsDone(cloudberry.parameters.timeInterval)) {
+          cloudberryClient.send(heatJson, function(id, resultSet, resultTimeInterval){
+            if(angular.isArray(resultSet)) {
+              cloudberry.commonTweetResult = resultSet[0].slice(0, queryUtil.defaultSamplingSize - 1);
+              cloudberry.heatmapMapResult = resultSet[0];
+              heatMapResultCache.putValues(cloudberry.heatmapMapResult, cloudberry.parameters.timeInterval)
+              cloudberry.heatMapMinDate = cloudberry.heatmapMapResult[cloudberry.heatmapMapResult.length-1]["create_at"];
+            }
+          }, "heatMapResult");
+        }
+        
+        // Complete heatmap cache hit case
+        else {
+          cloudberry.heatmapMapResult = heatMapResultCache.getValues(cloudberry.parameters.timeInterval);
+        }
       }
+    
       // Partial time series cache hit case
       else {
+        var heatTimeJson = queryUtil.getTimeBarRequest(cloudberry.parameters, $scope.geoIdsNotInTimeSeriesCache);
+      
+        cloudberryClient.send(heatJson, function(id, resultSet, resultTimeInterval) {
+          if(angular.isArray(resultSet)) {
+            cloudberry.commonTweetResult = resultSet[0].slice(0, queryUtil.defaultSamplingSize - 1);
+            cloudberry.heatmapMapResult = resultSet[0];
+            heatMapResultCache.putValues(cloudberry.heatmapMapResult, cloudberry.parameters.timeInterval)
```

Comment:
Here it proves the above conjecture, that we call `putValues` every time we get results from Cloudberry, and since we enabled the `delta` feature, then each time the results coming from Cloudberry have disjoint time range.